src/content/1.7/development/components/grid-component.md

```diff
@@ -0,0 +1,204 @@
+* The *Grid actions* are the related actions available for this grid: in PrestaShop it's common to have "export" or "access to sql manager" actions.
+* The *Row actions* are the related actions available for an entry of this list of data: in PrestaShop it's common to be able to edit/access or delete the entry
+* The *Bulk actions* are the actions available for a bunch of selectable entries: a bulk delete or a bulk edition for instance.
+
+You don't have to create the Grid Definition by yourself byt rely instead on a Grid Definition Factory.
+
+This factory must implements the `GridDefinitionFactoryInterface` interface which have only one method: `create()`.
+You'd better use the abstract class provided by the component, which give you access to the translator and already implement re-usable functions for you:
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\AbstractGridDefinitionFactory;
+use PrestaShop\PrestaShop\Core\Grid\Action\GridActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+final class FooGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    // required
+    protected function getIdentifier()
+    {
+        return 'foo';
+    }
+
+    protected function getName()
+    {
+        return $this->trans('Foo', [], 'Admin.Advparameters.Feature');
+    }
+
+    // required
+    protected function getColumns()
+    {
+        return ColumnCollection::fromArray([
+            [
+                'identifier' => 'id',
+                'name' => $this->trans('ID', [], 'Admin.Global'),
+                'filter_form_type' => TextType::class
+            ],
+            ...
+        ]);
+    }
+
+    // null by default
+    protected function getGridActions()
+    {
+        return GridActionCollection::fromArray([
+            [
+                'identifier' => 'delete',
+                'name' => $this->trans('Erase all', [], 'Admin.Advparameters.Feature'),
+                'icon' => 'delete_forever',
+                'renderer' => '<a href="foo/delete">Delete</a>',
+            ],
+            [
+                'identifier' => 'ps_refresh_list',
+                'name' => $this->trans('Refresh list', [], 'Admin.Advparameters.Feature'),
+                'icon' => 'refresh',
+            ],
+        ]);
+    }
+}
+```
+
+### The Grid Data Provider
+
+As you can imagine, the responsibility of Grid Data Provider is to provide the Grid data, from the Grid Definition and using the Grid Query Builder.
+
+The only method available of `GridDataProviderInterface` is `getData` which returns an instance of `GridData`.
+A GridData is an immutable object used to store and retrieve the GridData, so if you want to alter this data, you must do it in Grid Data Provider *before* the GridData creation.
+
+There is a good news here: you don't need to create your own as we provide one: the `GridDataProvider`. 
```

Comment:
`your own as we provide one: the` => `your own as we provide one, the:`?

Reply:
This will be updated soon, I'm working on it with @sarjon anyway